azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_base.py

```diff
@@ -736,32 +731,146 @@ def do_retry():
 ]
 
 
+class ReconnectState(object):
+    never_connected = "never_connected"
+    waiting_to_reconnect = "waiting_to_reconnect"
+    deferred = "deferred"
```

Comment:
>deferred = "deferred" [](start = 4, length = 21)

This is a strange name but I'm not sure what else to call it.  "deferred' means "defer the interpretation of the state to self.pipeline_root.connected." Maybe "whatever_the_pipeline_root says" is a better name.